src/ucs/datastruct/arbiter.c

```diff
@@ -126,33 +175,22 @@ void ucs_arbiter_group_purge(ucs_arbiter_t *arbiter,
     } while (ptr != tail);
 
     if (is_scheduled) {
-        if (orig_head == prev_group) {
-            /* this is the only group which was scheduled */
-            if (group->tail == NULL) {
+        if (group->tail == NULL) {
```

Comment:
can we revert this function just for v1.6.x branch (e.g don't use common code) - we will have some code duplication on v1.6.x only, but at least not change what is not absolutely required